jscomp/super_errors/super_misc.mli

```diff
@@ -0,0 +1,6 @@
+(** Range coordinates all 1-indexed, like for editors. Otherwise this code
+  would have way too many off-by-one errors *)
+val print_file: range:(int * int) * (int * int) -> lines:string array -> Format.formatter -> unit -> unit
+
+(** This will be called in super_main. This is how you override the default error and warning printers *)
+val colorize_tagged_string: Format.formatter -> (string -> Ext_color.style list) -> unit
```

Comment:
I'm not using it for warn/error etc. I'm using it like so: https://github.com/BuckleScript/bucklescript/pull/1857/files#diff-98efb70d9d6c67e672eaf4a1a529e112R44

This way these printing functions don't need to know whether they're emitting a warning, an error, or a themed color in the future (some colors don't work well at all on the white terminal theme or windows blue).

If you want a single one then I can gather these (including maybe a few dozens of https://github.com/BuckleScript/bucklescript/pull/1857/files#diff-4b8762e0b86bfe6187fbc1eca3f9d139R237) under a single switch?